harness/determined/tensorboard/metric_writers/callback.py

```diff
@@ -54,13 +62,36 @@ def on_train_step_end(
                 continue
             self._maybe_write_metric(name, value, steps_completed)
 
-        self.writer.reset()
+        self.writer.flush()
+        self._last_flush_ts = time.time()
 
     def on_validation_step_end(self, steps_completed: int, metrics: Dict[str, Any]) -> None:
         logger.debug("Write validation metrics for TensorBoard")
+
+        self._maybe_reset()
+
         for name, value in metrics.items():
             if not name.startswith("val"):
                 name = "val_" + name
             self._maybe_write_metric(name, value, steps_completed)
 
-        self.writer.reset()
+        self.writer.flush()
+        self._last_flush_ts = time.time()
+
+    def _maybe_reset(self) -> None:
+        """
+        Reset (close current file and open a new one) the current writer if more than 1 second
+        has elapsed since the last reset.
+
+        The TensorFlow event writer names each event file by the epoch second it is created, so
+        if events are written quickly in succession (< 1 second apart), they will overwrite each
+        other.
+
+        This effectively batches event writes so each event file may contain more than one event.
+        """
+        current_ts = time.time()
+        if not self._last_flush_ts:
+            return
+
+        if int(current_ts) > int(self._last_flush_ts):
+            self.writer.reset()
```

Comment:
why the different implementation and different docstring?

In `_pytorch_context.py`, the reset logic was entirely encapsulated in `_maybe_reset_tbd_writer()`, but here it's spread out?  Why is that?

Reply:
discussed offline, updated